tooling/helpers/rest_helper.py

```diff
@@ -18,28 +23,37 @@ def await_transcription_finish(id: str, api_key: str):
             if status["transcription_id"] != id:
                 continue
             if status["status"] not in ["in_progress", "in_query"]:
-                is_finished = True
+                return True
+        count+=1
+    print("Wait time for rest transcription was reached. This could be an error or your machine being slow (in the ladder case increase the limit)")
+    return False
 
 
-def transcribe_file_rest(filepath: str, api_key: str) -> str:
+def transcribe_file_rest(filepath: str, api_key: str, scale: str) -> RestResult:
+    result = RestResult(scale=scale)
+    start_time = time.time()
     r = None
     with open(filepath, "rb") as f:
         r = requests.post(
             "http://localhost:8393/transcriptions",
             files={"file": f},
             headers={"Authorization": api_key},
         )
-    if r.status_code != 200:
-        print(r.status_code)
-        print("Something went wrong")
-        return ""
+    if r.status_code != HTTPStatus.OK:
+        print(f"Unexpected HTTP response from REST: Wanted {HTTPStatus.OK} got {r.status_code}")
+        result.faulty = True
+        return result
     id = r.json()["transcription_id"]
-    await_transcription_finish(id, api_key)
+    if not await_transcription_finish(id, api_key):
+        result.faulty = True 
+        return result
     transcription_result = requests.get(
         f"http://localhost:8393/transcriptions/{id}",
```

Comment:
same

Reply:
Meant for local dev, i want to discourage running an 8 hour benchmark against the live instance